File: python/sentiment.py
from vaderSentiment.vaderSentiment import sentiment as vaderSentiment
import pymysql
import csv

connection = pymysql.connect(host='127.0.0.1',
                             user='root',
                             password='',
                             db='trump',
                             charset='utf8mb4',
                             cursorclass=pymysql.cursors.DictCursor)

with connection.cursor() as cursor:
    sql = "SELECT id, tweet " \
          "FROM tweets "
    cursor.execute(sql)
    result = cursor.fetchall()

p = {}
for tweet in result:
    vs = vaderSentiment(tweet['tweet'].encode('utf-8'))
    sql = "INSERT INTO twitter_sentiment_analysis (neg, neu, pos, compound, tweet_id) VALUES (%s, %s, %s, %s, %s)"
    with connection.cursor() as cursor:
        cursor.execute(sql, [vs['neg'], vs['neu'], vs['pos'], vs['compound'], tweet['id']])

    connection.commit()

with connection.cursor() as cursor:
    sql = "SELECT id, submission_title " \
          "FROM reddit "
    cursor.execute(sql)
    result = cursor.fetchall()

l = {}

for submission in result:
    vs = vaderSentiment(submission['submission_title'].encode('utf-8'))
    sql = "INSERT INTO reddit_sentiment_analysis (neg, neu, pos, compound, reddit_id) VALUES (%s, %s, %s, %s, %s)"
    with connection.cursor() as cursor:
        cursor.execute(sql, [vs['neg'], vs['neu'], vs['pos'], vs['compound'], submission['id']])

    connection.commit()

File: r/sentiment.r
list.of.packages <- c("RMySQL", "plotly")
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
if (length(new.packages)) install.packages(new.packages)
  
library(RMySQL)
library(plotly)

# Connect to the MySQL database
mydb = dbConnect(MySQL(), user='root', password='', dbname='trump', host='127.0.0.1')

# Query the data
rs = dbSendQuery(mydb, "select round(sum(tsa.pos), 2) as p, round(sum(tsa.neg), 2) as n from tweets as t join twitter_sentiment_analysis as tsa on t.id = tsa.tweet_id limit 0, 100000")
data = fetch(rs, n=-1)

# Create a bar chart
p <- plot_ly(
  x = c("Positive", "Negative"),
  y = c(as.numeric(data$p), as.numeric(data$n)),
  name = "Positive and Negative sum comparison.",
  type = "bar"
)

# Calculate the sentiment score for reddit submissions.
# Query the data
rs = dbSendQuery(mydb, "select round(sum(rsa.pos), 2) as p, round(sum(rsa.neg), 2) as n from reddit as r join reddit_sentiment_analysis as rsa on r.id = rsa.reddit_id;")
data = fetch(rs, n=-1)

# Create a bar chart
p <- plot_ly(
  x = c("Positive", "Negative"),
  y = c(as.numeric(data$p), as.numeric(data$n)),
  name = "Positive and Negative sum comparison.",
  type = "bar"
)

p

